sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -57,23 +58,30 @@ async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
 
         public async Task<ServiceBusTriggerMetrics> GetMetricsAsync()
         {
-            ServiceBusReceivedMessage message = null;
+            ServiceBusReceivedMessage activeMessage = null;
             string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
 
             try
             {
-                // Peek the first message in the queue without removing it from the queue
-                // PeekAsync remembers the sequence number of the last message, so the second call returns the second message instead of the first one
-                // Use PeekBySequenceNumberAsync with fromSequenceNumber = 0 to always get the first available message
-                message = await _receiver.Value.PeekMessageAsync(fromSequenceNumber: 0).ConfigureAwait(false);
+                var receivedMessages = await _receiver.Value.PeekMessagesAsync(10, fromSequenceNumber: lastSequenceNumber).ConfigureAwait(false);
```

Comment:
This differs from the logic in ScaleController (@chiangvincent made recent changes there for this issue). In that case we only ever peek the first message, and with Vincent's recent change, we won't use the message for determining QueueTime if it isn't Active. We don't loop through looking for an Active message.

I'm not necessarily saying moving forward on each iteration isn't the right thing to do - it'd just be different than what we do in ScaleController. We should have consistent behavior in both places I think.

Reply:
There is an active issue: https://github.com/Azure/azure-service-bus/issues/95 to update the SDK that will help peek at active message. @chiangvincent  lets discuss if the approach here should be carried into ScaleController as well.

Reply:
I'm happy either way to adjust this PR to make things consistent. I guess given the other code is not public it might be difficult for me to reach parity though

Reply:
FYI just looking at the state of one message was what I had initially proposed 😉

https://github.com/Azure/azure-sdk-for-net/pull/25127/commits/7da13947b039c5c5d43fa64656d0a4a5db138090

Reply:
Just to clarify if we send a static batch of 1000 deferred followed by 1000 active messages, we'd discover the first active message and start scaling after 16 min? Assuming a polling interval of 10 seconds: (1000 messages / 10 messages per batch * 10 seconds per batch / 60 seconds per min). It might be advantageous to be able to adjust the batch size used for iteration in this case, as this could cause significant delays in scaling up.

**Choosing 1000 messages as a threshold because one scale heuristic is keeping the ratio of messages to worker below 1000:1.

Reply:
See this discussion too https://github.com/Azure/azure-sdk-for-net/pull/25127#discussion_r756293126

The initial implementation just continued peeking until it found an active message or reached the end. 

https://github.com/Azure/azure-sdk-for-net/pull/25127/commits/a3dec639aae5c60ddb1678e6cbe5ca4d7cf665d8

It was suggested to go in another direction. Would you prefer to restore that behavior or a fundamentally different direction? As long as all involved people here are aligned, I'm happy to change things.

Reply:
Thanks for your patience Daniel. We discussed a bit on the team, and the algorithm you have here is good. If we don't think that we can do the lastSequenceNumber paging in a safe way, even looking at a single batch starting at 0 each time would be an improvement (you had that as well in a previous iteration I recall).

Reply:
So we have

- What we do currently in this PR
- Peeking until we find an active or reach the end
- Looking at a single batch by starting at 0 each time

My preference would be on the peeking until we find an active message or reach the end. I could also introduce a timed cancellation token source that keeps the peeking within a constraint time window. What do you think?

Reply:
I think for now we should keep it simple and just:

- peek one message, and if active return
- otherwise, peek a single batch and look for an active one
- if there were messages, but none active log a Debug level message

The log message will help us see how prevalent this is, and help us with support cases.

Reply:
Sorry for the delay. Took a while until I had some quiet leisure time to get this going

Reply:
I'm a bit confused as to where we landed here - it seems that the functions team agreed with the approach of incrementing the last peeked sequence each time, but in the latest implementation we always peek from the head of the queue.

Reply:
@mathewc can you clarify what the recommended approach is?

Reply:
Is what we agreed upon https://github.com/Azure/azure-sdk-for-net/pull/25127#discussion_r798772395

Reply:
but peek one message from where 😄

Reply:
Don't be the party crasher 😜😁😎

Reply:
@JoshLove-msft what we finally agreed upon after some internal discussions as well was the approach 7 comments above. It's a compromise attempting to keep things simple, while also providing relief for most common cases of a small number of inactive messages.

Reply:
Alright, works for me!